app/controllers/api/base_controller/parameters.rb

```diff
@@ -160,20 +160,24 @@ def sort_params(klass)
         orders = String(params['sort_order']).split(",")
         options = String(params['sort_options']).split(",")
         params['sort_by'].split(",").zip(orders).collect do |attr, order|
-          if klass.attribute_method?(attr) || klass.method_defined?(attr) || attr == klass.primary_key
-            sort_directive(attr, order, options)
+          if klass.virtual_attribute?(attr) && !klass.attribute_supported_by_sql?(attr)
```

Comment:
@kbrock would definitely be more simplistic, but I thought it might be a good idea to have both the `is not a valid attribute` and `cannot be sorted by` messages separate for clarification